### ISSUE TITLE
"keeping existing reason" does not work for some custom reasons in the advanced file deletion dialog

### ISSUE THREAD

**Issue by NazrinRat666** [hydrusnetwork/hydrus#1653]
### Hydrus version

v603

### Qt major version

Qt 6

### Operating system

Windows 10

### Install method

Extract

### Bug description and reproduction

![00-28-59](https://github.com/user-attachments/assets/d1937a7e-5021-4f44-9f53-44af8511a101)
![00-29-11](https://github.com/user-attachments/assets/5d105b45-2f30-4fcc-95cf-e70a97b20db6)

When using the advanced file deletion dialog on a file that's been deleted through a custom reason added in `options->files and trash->advanced file deletion and custom reasons`; "keep existing reason: (...)" is unchecked which is wrong and not consistent with custom reasons added in the advanced file deletion dialog.

**Comment by hydrusnetwork** [comment]
Thank you for this report. I do not have time for it this week, but I will get to it for next week and figure something out. This dialog is a mess behind the scenes and needs more cleanup and simplification.

**Comment by hydrusnetwork** [comment]
Done for tomorrow, here's the (dense, sorry!) changelog item atm:

`cleaned up the reason-initialisation of the advanced file deletion dialog. the list of reasons is now much more item-position-stable and will not create duplicates when intersecting options clash. the selection rules are now simple: if all the file(s) have existing deletion reasons, this is listed, marked as the existing reason, and selected; otherwise, if the dialog is set to remember the previous deletion reason, this is selected; otherwise the "default" reason the dialog launches with is selected. the thing that is selected has a more stable position as the list now always follows an order of (optional unique default reason, list of reasons, optional unique existing reason, optional do-not-alter existing reason, custom reason), and items are marked in-place if they are interesting`

Basically this thing now has a better workflow and will just feel better to use, I hope! Let me know how you get on with it.
